### PR TITLE
[ARCTIC-1231] Add a timeout strategy to avoid blocking the UTs ArcticSourceTest

### PR DESCRIPTION
**Pull request by YesOrNo828**
## Why are the changes needed?

fix: #1231  
## Brief change log

  - *Add a timeout strategy to the ArcticSourceTest in the flink modules*
  - *The flink 1.12, 1.14, and 1.15 versions are affected.*
  - *Add some logs to help locate the problem*
  - *Reduce the processing time during executing the ArctciSourceTest UTs*

## How was this patch tested?
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no) no 
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) no

### CODE REVIEW

File: flink/v1.15/flink/src/test/java/com/netease/arctic/flink/read/ArcticSourceTest.java [hunk 1]
```diff
@@ -394,7 +394,7 @@ public void testArcticSourceEnumeratorWithChangeExpired() throws Exception {
     LOG.info("commit empty snapshot");
     AppendFiles changeAppend = table.changeTable().newAppend();
     changeAppend.commit();
-    Thread.sleep(ScanContext.MONITOR_INTERVAL.defaultValue().toMillis() * 2);
+    Thread.sleep(MONITOR_INTERVAL.defaultValue().toMillis() * 2);
```
Comment (lklhdu) [inline flink/v1.15/flink/src/test/java/com/netease/arctic/flink/read/ArcticSourceTest.java:397]: Does this unit test also need to use the initArcticSourceWithMonitorInterval function?

Comment (YesOrNo828) [reply flink/v1.15/flink/src/test/java/com/netease/arctic/flink/read/ArcticSourceTest.java:399]: Good catch.

**Review by lklhdu — APPROVED** [review]
LGTM.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**